libft/sources/map/mapss/ft_mapss_finalize.c

```diff
@@ -6,17 +6,40 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/08/13 23:12:55 by inikulin          #+#    #+#             */
-/*   Updated: 2024/10/23 17:54:21 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/27 00:34:13 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "mapss.h"
 
-// implement!
-int	ft_mapss_finalize(t_mapss *map, int mode, char *msg, int retval)
+int	ft_mapss_finalize_i(t_mapss *map, char *msg, int retval)
 {
-	(void)map;
-	(void)mode;
-	(void)msg;
+	t_dlist			*node;
+	t_dlist			*prev;
+	t_mapss_entry	*entry;
+
+	if (map)
+	{
+		node = map->head;
+		while (node)
+		{
+			entry = node->content;
+			ft_free_s_null((void **)&entry->key);
+			ft_free_s_null((void **)&entry->value);
+			free(entry);
+			prev = node;
+			node = node->next;
+			free(prev);
```

Comment:
here you will free node twice if node is NULL. not mistake, but redundant and you can do instead:
```
t_dlist *next;
next = node->next;
free(node); 
```
maybe it's clearer to read also.

Reply:
I'm not sure I see the double free... if node is null, we just don't go into the loop?..

Reply:
tou are right, my mistake, the name prev confused me. cur would be clearer as we stell work with this not, prev is confusing. but ok, up to you to change it. when you read, you can resolve and merge